aspnetcore/blazor/components/lifecycle.md

```diff
@@ -51,14 +51,41 @@ Developer calls to [`StateHasChanged`](#state-changes) result in a render. For m
 
 ### Before parameters are set
 
-<xref:Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync%2A> sets parameters supplied by the component's parent in the render tree:
+<xref:Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync%2A> sets parameters supplied by the component's parent in the render tree or from route parameters. By overriding the method, developer code can interact directly with the <xref:Microsoft.AspNetCore.Components.ParameterView>'s parameters.
 
-```csharp
-public override async Task SetParametersAsync(ParameterView parameters)
-{
-    await ...
+In the following example, <xref:Microsoft.AspNetCore.Components.ParameterView.TryGetValue%2A?displayProperty=nameWithType> assigns the `Param` parameter's value to `value` if parsing a route parameter for `Param` is successful. When `value` isn't `null`, the value is displayed by the `SetParametersAsyncExample` component.
+
+`Pages/SetParametersAsyncExample.razor`:
+
+```razor
+@page "/setparametersasync-example/{Param?}"
+
+<h1>SetParametersAsync Example</h1>
+
+<p>@message</p>
+
+@code {
+    private string message;
 
-    await base.SetParametersAsync(parameters);
+    [Parameter]
+    public string Param { get; set; }
+
+    public override async Task SetParametersAsync(ParameterView parameters)
+    {
+        if (parameters.TryGetValue<string>(nameof(Param), out var value))
+        {
+            if (value is null)
```

Comment:
Instead of doing this if statement, you can do something like:

```csharp
if (!parameters.TryGetValue<string>(nameof(Param), out var value))
{
  message = $"The value of 'Param' is null.";
}
else 
{
  message = $"The value of 'Param' is {value}.";
}
```